### PR TITLE
Expand env variables for cmds/entrypoint with format $(ENV)

### PR DESCRIPTION
**Pull request by rhatdan**
Kubernetes supports expanding $(FOOBAR) as environment variables within
the kube.YAML. When using podman kube play, we need to do the same, for
supporting these YAML files.

Fixes: #15983

#### Does this PR introduce a user-facing change?

```release-note
Commands and entrypoints can use environment variables.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
Replaces: https://github.com/containers/podman/pull/18881
@ygalblum PTAL

**Review by ygalblum — APPROVED** [review]
LGTM

**Comment by umohnani8** [comment]
changes LGTM

File: test/e2e/play_build_test.go [hunk 1]
```diff
@@ -100,6 +140,44 @@ LABEL marge=mom
...
+
+		session := podmanTest.Podman([]string{"play", "kube", "echo.yaml"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		inspect := podmanTest.Podman([]string{"container", "inspect", "echo_pod-foobar"})
+		inspect.WaitWithDefaultTimeout()
+		Expect(inspect).Should(Exit(0))
+		inspectData := inspect.InspectContainerToJSON()
+		Expect(inspectData).ToNot(BeEmpty())
```
Comment (edsantiago) [inline test/e2e/play_build_test.go:172]: Suggestions:
- check container logs, confirm that output == "`BAR BAR \$\(FOO\) FOOBARBAZ`"
- add more args, like `$(BAZ) ${BAZ}`, *without* a corresponding `env` yaml, but *with* `Putenv()` in the test.

**Comment by TomSweeneyRedHat** [comment]
nothing over @edsantiago 's comments.  Tests are very red.

File: go.mod [hunk 2]
```diff
@@ -71,6 +71,7 @@ require (
 	google.golang.org/protobuf v1.31.0
 	gopkg.in/inf.v0 v0.9.1
 	gopkg.in/yaml.v3 v3.0.1
+	k8s.io/kubernetes v1.13.0
```
Comment (flouthoc) [inline go.mod:74]: Afaik k8s deps were removed intentionally, this PR is adding it back. @Luap99 @umohnani8 PTAL

**Review by flouthoc — CHANGES_REQUESTED** [review]
Comment https://github.com/containers/podman/pull/19630/files#r1299644346 needs to be ackd by maintainers.

File: go.mod (hunk 2 above)
Comment (vrothberg) [reply go.mod:74]: It's only adding k8s.io/kubernetes/third_party/forked/golang/expansion as dependency which I think is tolerable.

**Review by vrothberg — COMMENTED** [review]
Please add a commit message.

File: go.mod (hunk 2 above)
Comment (Luap99) [reply go.mod:74]: The K8s deps were only dropped because they added ridiculous amount of bloat that was never used in podman, i.e. the full RPC code. The code imported into podman (pkg/k8s.io) is heavily modified, mainly dropped all the generated protobuf code that was responsible for most of the binary size.

If this is only using a single package without big dependencies it is totally fine to use that. And I much rather prefer that in order to get potential updates.

**Comment by rhatdan** [comment]
I agree that adding `k8s.io/kubernetes` potentially opens us to Bloat, but for now this is only using the expansion subdir, so no bloat.

**Comment by vrothberg** [comment]
I think @ygalblum brought up another issue though

> BTW if we do add the dependency back, why are we taking such an old version?

Is this package not available in newer K8s versions?

**Review by vrothberg — APPROVED** [review]
LGTM

**Comment by ygalblum** [comment]
LGTM

File: test/e2e/play_build_test.go [hunk 3]
```diff
@@ -278,4 +278,81 @@ LABEL marge=mom
...
+
+		Expect(len(inspectData[0].Args)).To(BeEquivalentTo(5))
+		Expect(inspectData[0].Args[1]).To(BeEquivalentTo("echo BAR"))
+		Expect(inspectData[0].Args[2]).To(BeEquivalentTo("${FOO}"))
+		Expect(inspectData[0].Args[3]).To(BeEquivalentTo("$(FOO)"))
+		Expect(inspectData[0].Args[4]).To(BeEquivalentTo("FOOBARBAZ"))
```
Comment (edsantiago) [inline test/e2e/play_build_test.go:355]: I think something is wrong. I took the liberty of adding the test I requested:
```go
		logs := podmanTest.Podman([]string{"logs", "echo_pod-foobar"})
		logs.WaitWithDefaultTimeout()
		Expect(logs).Should(Exit(0))

		Expect(logs.OutputToString()).To(Equal("BAR BAR BAR FOOBARBAZ"))
```
...and it's failing:
```
[FAILED] Expected                                                                                                                                             
      <string>: BAR BAR BAR BAR BAR BAR BAR BAR                                                                                                                 
  to equal                                                                                                                                                      
      <string>: BAR BAR BAR FOOBARBAZ
```
It's impossible to know which BAR corresponds to what, but I really don't like that failure. So, sorry, placing hold until someone explains it or fixes it.

File: pkg/domain/entities/events.go [hunk 4]
```diff
@@ -62,8 +63,11 @@ func ConvertToEntitiesEvent(e libpodEvents.Event) *Event {
 	}
 	attributes["image"] = e.Image
 	attributes["name"] = e.Name
+	fmt.Println("DAN", e.ContainerExitCode)
 	attributes["containerExitCode"] = strconv.Itoa(e.ContainerExitCode)
-	attributes["podId"] = e.PodID
+	if e.PodID != "" {
+		attributes["podId"] = e.PodID
+	}
```
Comment (Luap99) [inline pkg/domain/entities/events.go:70]: needs to be removed as this should not be part of this PR

**Review by Luap99 — CHANGES_REQUESTED** [review]

**Comment by edsantiago** [comment]
Tests LGTM, thank you for accepting my suggestion.

**Review by Luap99 — APPROVED** [review]
LGTM
